internal/backend_helpers/snmpSwitch.go

```diff
@@ -0,0 +1,501 @@
+	}
+
+	slsURL, ok := os.LookupEnv("SLS_URL")
+	if !ok {
+		log.Fatal("Environment variable slsURL was not set")
+	}
+
+	serviceName, reqErr := base.GetServiceInstanceName()
+	if reqErr != nil {
+		serviceName = "RTS-SNMP"
+		log.WithField("err", reqErr).Infof("WARNING, can't get service/instance name, using '%s'", serviceName)
+	}
+	helper.SLS = slsapi.NewSLS(slsURL, httpClient, serviceName)
+
+	// Time to setup the connection to Vault
+	rtsVaultKeypath, ok := os.LookupEnv("JAWS_VAULT_KEYPATH")
+	if !ok {
+		log.Fatal("Environment variable JAWS_VAULT_KEYPATH was not set")
+	}
+
+	secureStorage := connectToVault()
+	if secureStorage == nil {
+		log.Fatal("Unable to connect to Vault")
+	}
+
+	helper.RTSCredentialStore = rts_credential_store.NewRTSCredStore(rtsVaultKeypath, secureStorage)
+
+	if ok := setupCompCredsVault(secureStorage); !ok {
+		log.Fatal("Failed to setup CompCredsVault")
+	}
+
+	// Get default credentails from vault
+	creds, credErr := helper.RTSCredentialStore.GetGlobalCredentials()
+	if credErr != nil {
+		err = credErr
+		log.WithField("err", err).Fatal("Unable to get default credentails for RTS")
+	}
+
+	password := creds.Password
+	username := creds.Username
+	role := "Administrator"
+	err = helper.RedisHelper.initAccount(username, role, password)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":    err,
+			"username": username,
+			"role":     role,
+		}).Error("problem encountered initializing account")
+		return
+	}
+
+	// Certificate Service setup
+	err = helper.CertificateService.SetupCertificateService("RTS", GenericCertificateID)
+	if err != nil {
+		log.WithError(err).Error("Failed to setup certificate service")
+		return
+	}
+
+	return
+}
+
+// Get Environment for backend handler invocations
+func (helper *SNMPSwitchHelper) GetEnvForXname(xname string) (env map[string]string, err error) {
+	env = map[string]string{}
+
+	if snmpDevice, ok := helper.KnownDevices[xname]; ok {
+		env["RTS_XNAME"] = snmpDevice.Xname
+
+		return
+	}
+
+	err = errors.New("Unable to find xname in device list")
+	log.WithField("xname", xname).Error(err)
+
+	return
+}
+
+// Called for every request a backend might be able to service.
+func (helper *SNMPSwitchHelper) RunBackendHelper(ctx context.Context, key string, args []string, env map[string]string) (value string, err error) {
+
+	xname, ok := env["RTS_XNAME"]
+	if !ok {
+		err = errors.New("RTS_XNAME not in environment variables")
+		log.Error(err.Error())
+		return
+	}
+
+	// Check to make sure we actually know about this device.
+	snmpDevice, deviceIsKnown := helper.KnownDevices[xname]
+	if !deviceIsKnown {
+		err = fmt.Errorf("unknown xname: %s", xname)
+		log.Error(err.Error())
+		return
+	}
+
+	// Do some sanity checks to make sure we're getting what we expect.
+	// Every key that comes in could be xname prefixed, use a regular expression to capture that.
+	keyRegex := regexp.MustCompile(`(x[0-9]{1,4}c[0-7]w[1-9][0-9]*)*(` + ServiceRootKeyspace + `\S+$)`)
```

Comment:
I probably don't understand.
Could the last `*` should be a `?` instead?
i.e. this:
`(x[0-9]{1,4}c[0-7]w[1-9][0-9]*)?(`
instead of:
`(x[0-9]{1,4}c[0-7]w[1-9][0-9]*)*(`

I'm assuming that:
`*` means match zero to many
`?` means match zero or one

Reply:
It probably should just be `?`. The other backend_helpers do it this way with the `*`. I'll look into whether or not it needs to be that way for some odd reason.